core/src/main/java/com/graphhopper/routing/util/parsers/FootPriorityParser.java

```diff
@@ -104,8 +104,20 @@ void collect(ReaderWay way, TreeMap<Double, PriorityCode> weightToPrioMap) {
         if (way.hasTag("foot", "designated"))
             weightToPrioMap.put(100d, PREFER);
 
-        if (way.hasTag("foot", "use_sidepath")) {
-            weightToPrioMap.put(100d, VERY_BAD); // see #3035
+        if (way.hasTag("foot", "use_sidepath") || way.hasTag("sidewalk", "separate") || way.hasTag("sidewalk:both", "separate")) {
```

Comment:
As all the added if statements below lead to the same result I would prefer to have a singe if statement with a bunch of logical operations instead.